### PR TITLE
Fixed #25085 -- Overrode Select Widget's __deepcopy__

### PR DESCRIPTION
**Pull request by ericfc**

### CODE REVIEW

**Comment by timgraham** [comment]
buildbot, add to whitelist.

File: tests/forms_tests/tests/test_widgets.py [hunk 1]
```diff
@@ -1997,3 +1997,19 @@ def test_form_label_association(self):
...
+
+    def test_deepcopy(self):
+        """
+        Ensure that __deepcopy__ copies all fields properly.
+        Refs #25085
+        """
+        widget = Select()
+        obj = copy.deepcopy(widget)
+        self.assertTrue(widget is not obj)
+        self.assertTrue(widget.choices == obj.choices)
```
Comment (timgraham) [inline tests/forms_tests/tests/test_widgets.py]: is there a reason not to use `assertEqual`?

File: tests/forms_tests/tests/test_widgets.py [hunk 2]
```diff
@@ -1997,3 +1997,19 @@ def test_form_label_association(self):
         # label tag is correctly associated with first rendered dropdown
         a = GetDate({'mydate_month': '1', 'mydate_day': '1', 'mydate_year': '2010'})
         self.assertIn('<label for="id_mydate_day">', a.as_p())
+
+
+class SelectWidgetTests(SimpleTestCase):
+
+    def test_deepcopy(self):
+        """
+        Ensure that __deepcopy__ copies all fields properly.
```
Comment (timgraham) [inline tests/forms_tests/tests/test_widgets.py]: fields -> attributes?

File: tests/forms_tests/tests/test_widgets.py (hunk 1 above)
Comment (ericfc) [inline tests/forms_tests/tests/test_widgets.py]: I wanted to make the difference between the two comparisons clear. Do you think it should be changed?

Comment (timgraham) [inline tests/forms_tests/tests/test_widgets.py]: Yes, in case of a failure it gives an easier to debug message.

**Comment by timgraham** [comment]
merged in 8ee6a3f1a855bf983639a14fc2393baa8ead741f, thanks!

**Pull request closed without merging**